src/cargo/core/compiler/custom_build.rs

```diff
@@ -249,6 +249,28 @@ fn build_work(cx: &mut Context<'_, '_>, unit: &Unit) -> CargoResult<Job> {
         }
     }
 
+    // Also inform the build script of the rustc compiler context.
+    if let Some(wrapper) = bcx.rustc().wrapper.as_ref() {
+        cmd.env("RUSTC_WRAPPER", wrapper);
+    }
```

Comment:
For explicitness could this `.env_remove("RUSTC_WRAPPER")` in the `else` case?

Reply:
I don't _think_ that should ever happen, since if `RUSTC_WRAPPER` is set in the environment, Cargo will presumably use that wrapper, and thus we're in the `if` case. But added for completeness 👍